mixer/adapter/solarwinds/internal/appoptics/batching.go

```diff
@@ -15,20 +15,27 @@
 package appoptics
 
 import (
+	"sync"
 	"time"
 
 	"istio.io/istio/mixer/pkg/adapter"
 )
 
+// CloseKey defines the key for synchronized map value for loopFactor.
+const CloseKey = "if-close"
+
 // BatchMeasurements reads slices of Measurement types off a channel populated by the web handler
 // and packages them into batches conforming to the limitations imposed by the API.
-func BatchMeasurements(loopFactor *bool, prepChan <-chan []*Measurement,
+func BatchMeasurements(loopFactor *sync.Map, prepChan <-chan []*Measurement,
```

Comment:
Could this be handled by just reading *loopFactor with an atomic read? Using a map as a communication mechanism seems strange.

Reply:
I think sync.Map is idiomatic solution in go for synchronized data variables. If it's a complex structure, i would go for implementing locking pattern using mutex myself. For this particular case, using this language feature gives the cleanest fix.

Reply:
You should use sync.atom.Value.
sync.Map is not idiomatic here.